### PR TITLE
k8s binary versions - move to manifest.cue and reduce repetition in build test

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind feature
/kind cleanup

**What this PR does / why we need it**:
Needed for: 
1. k8s version automation
2. 1 k8s version per VHD work

This moves k8s binary versions from two bash scripts (install-dependencies.sh and linux-vhd-content-test.sh), to a single config file (manifest.cue). 
Reducing repetition of versions makes k8s versioning faster and moving the versions to a config file, provides a way to automate k8s versioning (and also splitting k8s versions amongst VHDs).

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by cameronmeissner — APPROVED** [review]

File: vhdbuilder/packer/install-dependencies.sh [hunk 1]
```diff
@@ -462,18 +462,8 @@ apt-get -y clean
...
-1.23.8-hotfix.20220620
-1.23.12
-1.24.3
-1.24.6
-1.25.2-hotfix.20221006
-"
-
-KUBE_BINARY_VERSIONS="${MULTI_ARCH_KUBE_BINARY_VERSIONS}"
+# regular version >= v1.17.0 or hotfixes >= 20211009 has arm64 binaries. 
+KUBE_BINARY_VERSIONS="$(jq .kubernetes.versions parts/linux/cloud-init/artifacts/manifest.json | jq -r ".[]")"
```
Comment (alexeldeib) [inline vhdbuilder/packer/install-dependencies.sh]: nit: `jq .kubernetes.versions[]` ?

File: schemas/manifest.cue [hunk 2]
```diff
@@ -83,9 +83,18 @@
         "versions": [],
     },
     "kubernetes": {
-        "fileName": "",
+        "fileName": "kubernetes-node-linux-arch.tar.gz",
         "downloadLocation": "",
         "downloadURL": "https://acs-mirror.azureedge.net/kubernetes/v${PATCHED_KUBE_BINARY_VERSION}/binaries/kubernetes-node-linux-${CPU_ARCH}.tar.gz"
+        "versions": [
```
Comment (alexeldeib) [inline schemas/manifest.cue:89]: suggest to expand the schema above, similar to `containerd_ver`, can add `kubernetes_ver ~= [0-9]+.[0-9]+.[0-9]+(-hotfix.[0-9]{8})?` (think this matches all versions)

File: vhdbuilder/packer/test/linux-vhd-content-test.sh [hunk 3]
```diff
@@ -229,15 +230,7 @@ testKubeBinariesPresent() {
...
-  k8sVersions="
-  1.22.11-hotfix.20220620
-  1.22.15
-  1.23.8-hotfix.20220620
-  1.23.12
-  1.24.3
-  1.24.6
-  1.25.2-hotfix.20221006
-  "
+  k8sVersions="$(jq .kubernetes.versions $MANIFEST_FILEPATH | jq -r ".[]")"
```
Comment (alexeldeib) [inline vhdbuilder/packer/test/linux-vhd-content-test.sh:233]: `jq -r .kubernetes.versions[] $MANIFEST_FILPATH` ?

File: schemas/manifest.cue (hunk 2 above)
Comment (alexeldeib) [reply schemas/manifest.cue:89]: or not, if it's not actually a contract, just helps prevent manual errs.

Comment (ganeshkumar5699) [reply schemas/manifest.cue:95]: Updated it

File: vhdbuilder/packer/test/linux-vhd-content-test.sh (hunk 3 above)
Comment (ganeshkumar5699) [reply vhdbuilder/packer/test/linux-vhd-content-test.sh]: Updated

File: vhdbuilder/packer/test/linux-vhd-content-test.sh [hunk 4]
```diff
@@ -229,15 +230,7 @@ testKubeBinariesPresent() {
...
-  k8sVersions="
-  1.22.11-hotfix.20220620
-  1.22.15
-  1.23.8-hotfix.20220620
-  1.23.12
-  1.24.3
-  1.24.6
-  1.25.2-hotfix.20221006
-  "
+  k8sVersions="$(jq -r .kubernetes.versions[] --monochrome-output --compact-output < /opt/azure/manifest.json)"
```
Comment (alexeldeib) [inline vhdbuilder/packer/test/linux-vhd-content-test.sh:233]: is --compact-output necessary here? why not in install-deps if so?

**Review by alexeldeib — APPROVED** [review]

File: vhdbuilder/packer/test/linux-vhd-content-test.sh (hunk 4 above)
Comment (ganeshkumar5699) [reply vhdbuilder/packer/test/linux-vhd-content-test.sh:233]: It's not necessary. It was used in a couple of other places and I added it while debugging another issue. Can remove it.

**Pull request merged**